packages/web-app-pdf-viewer/src/App.vue

```diff
@@ -1,80 +1,27 @@
 <template>
-  <main>
-    <app-top-bar v-if="!loading && !loadingError" :resource="resource" @close="closeApp" />
-    <loading-screen v-if="loading" />
-    <error-screen v-else-if="loadingError" />
-    <div v-else class="oc-height-1-1">
-      <object class="pdf-viewer oc-height-1-1 oc-width-1-1" :data="url" type="application/pdf" />
-    </div>
-  </main>
+  <AppWrapper
+    v-slot="{ url }: AppWrapperSlotArgs"
+    application-id="pdf-viewer"
+    :url-for-resource-options="{
+      disposition: 'inline'
+    }"
+  >
+    <object class="pdf-viewer oc-width-1-1 oc-height-1-1" :data="url" type="application/pdf" />
+  </AppWrapper>
 </template>
+
 <script lang="ts">
 import { defineComponent } from 'vue'
-import { useAppDefaults } from 'web-pkg/src/composables'
-import AppTopBar from 'web-pkg/src/components/AppTopBar.vue'
-import ErrorScreen from './components/ErrorScreen.vue'
-import LoadingScreen from './components/LoadingScreen.vue'
+import AppWrapper, { AppWrapperSlotArgs } from 'web-pkg/src/components/AppTemplates/AppWrapper.vue'
 
 export default defineComponent({
   name: 'PDFViewer',
   components: {
-    AppTopBar,
-    ErrorScreen,
-    LoadingScreen
-  },
-  setup() {
-    return {
-      ...useAppDefaults({
-        applicationId: 'pdf-viewer'
-      })
-    }
-  },
-  data: () => ({
-    loading: true,
-    loadingError: false,
-    url: '',
-    resource: null
-  }),
-  watch: {
-    currentFileContext: {
-      handler: function () {
-        this.unloadPdf()
-        this.loadPdf(this.currentFileContext)
-      },
-      immediate: true
-    }
-  },
-  beforeUnmount() {
-    this.unloadPdf()
-  },
-  methods: {
-    async loadPdf(fileContext) {
-      try {
-        this.loading = true
-        const resource = await this.getFileInfo(fileContext)
-
-        if (resource.mimeType !== 'application/pdf') {
-          return
-        }
```

Comment:
Should we check for supported mimetypes in AppWrapper?

Reply:
Let's tackle this when porting the Preview-app?